index.js

```diff
@@ -4030,6 +4037,83 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  // TODO: Separate this in a diferent PR
+  //# compare :: a -> a -> Ordering
+  //.
+  //. Compares two values by JavaScript comparison operators and
+  //. returns an Ordering value.
+  //.
+  //. ```javascript
+  //. > S.compare(2, 3)
+  //. 'LT'
+  //. > S.compare(3, 2)
+  //. 'GT'
+  //. > S.compare(3, 3)
+  //. 'EQ'
+  //. ```
+  function compare(a1, a2) {
+    if (a1 < a2) {
+        return 'LT';
+    }
+    if (a1 > a2) {
+        return 'GT';
+    }
+    return 'EQ';
+  }
+  S.compare =
+  def('compare', {}, [a, a, Ordering], compare);
+
+  // TODO: Separate this in a diferent PR
+  //# insertBy :: (Applicative f, Foldable f, Monoid f) => (a -> a -> Ordering) -> a -> f a -> f a
+  //.
+  //. TODO: Write the documentation
+  //.
+  //. ```javascript
+  //. > S.insertBy(S.compare, 6, [4, 2, 7, 5])
+  //. [4, 2, 6, 7, 5]
+  //. ```
+  function insertBy(cmp, x, ys) {
+    if (ys.length === 0) {
+        return S.prepend(x, ys);
+    }
+    var y = S.head(ys).value;
```

Comment:
Gotcha, waiting for feedback here before I go in direct index direction: https://github.com/sanctuary-js/sanctuary/pull/365#discussion_r107914151